lib/src/Base/Geom/LevelSet.cxx

```diff
@@ -125,6 +128,50 @@ void LevelSet::setLevel(const NumericalScalar level)
   level_ = level;
 }
 
+/* Lower bound of the bounding box */
+NumericalPoint LevelSet::getLowerBound() const
+{
+  const UnsignedInteger dimension(getDimension());
+  NumericalPoint lowerBound(dimension);
+  LinearNumericalMathFunction translate(NumericalPoint(1, level_), NumericalPoint(1), IdentityMatrix(1));
+  NumericalMathFunction equality(translate, function_);
+  for (UnsignedInteger i = 0; i < dimension; ++i)
+    {
+      Matrix m(1, dimension);
+      m(0, i) = 1.0;
+      LinearNumericalMathFunction coordinate(NumericalPoint(dimension), NumericalPoint(1), m);
+      OptimizationProblem problem(coordinate, equality, NumericalMathFunction(), Interval());
+      problem.setMinimization(true);
+      Cobyla solver(problem);
```

Comment:
@dbarbier There is no clean way to specify an optimization algorithm through ResourceMap. You provide either a string or an integer, and the mapping has to be done inside of the code in a static way (see eg GeneralizedLinearModelAlgorithm). The only way I found when working on the KrigingAlgorithm was to give access to the optimization problem to allow the user to solve it with its own solver, then to reinject the result through one of the constructors.

Do you have in mind an alternative way to do that? If not, I prefer to hardcode the algorithm waiting for a better management of complex types in ResourceMap.